### PR TITLE
Lint and gofmt the gateway as part of verify

### PR DESCRIPTION
**Pull request by tylercloke**
Signed-off-by: Tyler Cloke <tylercloke@gmail.com>

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: api/scripts/swagger.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ func main() {
 			reader := bufio.NewReader(f)
 			writer := bufio.NewWriter(out)
 
-			_, _ = writer.WriteString("package api\n\nfunc init() {\n")
-			_, _ = writer.WriteString("	Swagger.Add(\"" + name + "\", `")
+			_, _ = writer.WriteString("package api\n\nfunc init() {\n")   // nolint: errcheck
+			_, _ = writer.WriteString("	Swagger.Add(\"" + name + "\", `") // nolint: errcheck
```
Comment (srenatus) [inline api/scripts/swagger.go]: 💭 if we add the `// nolint` comment, can't we just write 
```suggestion
			writer.WriteString("	Swagger.Add(\"" + name + "\", `") // nolint: errcheck
```

File: components/automate-gateway/api/cfgmgmt.pb.swagger.go [hunk 2]
```diff
@@ -436,15 +436,15 @@ func init() {
           "description": "Repeated field of dynamically typed values."
         }
       },
-      "description": "` + "`" + `ListValue` + "`" + ` is a wrapper around a repeated field of values.\n\nThe JSON representation for ` + "`" + `ListValue` + "`" + ` is JSON array."
```
Comment (srenatus) [inline components/automate-gateway/api/cfgmgmt.pb.swagger.go]: ⚠️ These changes shouldn't be committed, unless we fix the code that generates them.

File: components/automate-gateway/api/chef.pb.swagger.go [hunk 3]
```diff
@@ -177,15 +177,15 @@ func init() {
           "description": "Repeated field of dynamically typed values."
         }
       },
-      "description": "` + "`" + `ListValue` + "`" + ` is a wrapper around a repeated field of values.\n\nThe JSON representation for ` + "`" + `ListValue` + "`" + ` is JSON array."
```
Comment (srenatus) [inline components/automate-gateway/api/chef.pb.swagger.go]: same here

File: components/automate-gateway/api/scripts/swagger.go [hunk 4]
```diff
@@ -49,13 +49,13 @@ func main() {
 			if isPrefix || err != nil {
 				logrus.Fatalf("Failed with isPrefix=%t err=%v", isPrefix, err)
 			}
-			writer.WriteString(strings.Replace(string(line), "`", "` + \"`\" + `", -1))
-			writer.WriteString("\n")
+			writer.WriteString(strings.Replace(string(line), "`", "` + \"`\" + `", -1)) // nolint: errcheck
```
Comment (srenatus) [inline components/automate-gateway/api/scripts/swagger.go]: 😕 Is this script the same as the one above? What is going on here? ❓ (Not part of your PR, just confused)

File: components/automate-gateway/gateway/middleware/auth_interceptor.go [hunk 5]
```diff
@@ -165,24 +165,6 @@ func (a *authInterceptor) UnaryServerInterceptor() grpc.UnaryServerInterceptor {
 	}
 }
 
-func getProjectsFromMetadata(projectHeaderEntries []string) []string {
```
Comment (srenatus) [inline components/automate-gateway/gateway/middleware/auth_interceptor.go]: Oops thanks for the cleanup 👍

File: components/automate-gateway/gateway/middleware/auth_interceptor_internal_test.go [hunk 6]
```diff
@@ -74,43 +74,6 @@ func TestHeaderAuthValidForClientAndPeer(t *testing.T) {
 	}
 }
 
-func TestGetProjectsFromMetadata(t *testing.T) {
```
Comment (srenatus) [inline components/automate-gateway/gateway/middleware/auth_interceptor_internal_test.go]: Heh I wasn't aware we had these tests. They're a bit different from the ones I've created when refactoring this helper: https://github.com/chef/automate/blob/c435118f939f11e1ed09a30d6f4edab24b6b1aa4/lib/grpc/auth_context/auth_context_test.go#L139
😃 It's fine though.

File: components/automate-gateway/gateway/services.go [hunk 7]
```diff
@@ -344,6 +344,8 @@ type ProfileRequest struct {
 }
 
 func (s *Server) ProfileCreateHandler(w http.ResponseWriter, r *http.Request) {
+	// TODO (tc): This ctx is getting overriden by the authRequest call below and is never used.
+	// nolint: ineffectual
 	ctx, cancel := context.WithCancel(r.Context())
```
Comment (srenatus) [inline components/automate-gateway/gateway/services.go]: As mentioned in slack, these two lines could go away. It's ok if you don't want to add that to this PR. If you'd want to, that would be cool, too.

File: components/automate-gateway/handler/iam/v2beta/policy/policy.go [hunk 8]
```diff
@@ -242,7 +242,13 @@ func (p *Server) GetPolicyVersion(
 	}
 
 	major, err := versionFromInternalVersion(resp.Version.Major)
+	if err != nil {
+		return nil, err
+	}
```
Comment (srenatus) [inline components/automate-gateway/handler/iam/v2beta/policy/policy.go]: haha nice

File: api/scripts/swagger.go (hunk 1 above)
Comment (tylercloke) [reply api/scripts/swagger.go]: Nope, it'll still complain that the function returns type `Error` that we aren't checking.

File: components/automate-gateway/api/scripts/swagger.go (hunk 4 above)
Comment (tylercloke) [reply components/automate-gateway/api/scripts/swagger.go]: LOL no idea actually. Hmmmmmm.

File: api/scripts/swagger.go (hunk 1 above)
Comment (srenatus) [reply api/scripts/swagger.go]: OK, this is confusing. A) How does [this file pass then](https://github.com/chef/automate/pull/608/files#diff-a099af9c4237c8d4028c5612518211c0R52)? B) Why is this, not living in `components/automate-gateway` even included in linting?

File: components/automate-gateway/gateway/services.go [hunk 9]
```diff
@@ -510,14 +504,11 @@ func (s *Server) ProfileTarHandler(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Content-Length", contentLength)
 		w.Header().Set("Content-Type", "application/x-gzip")
 		w.Header().Set("Accept-Ranges", "bytes")
-		w.Write(data.GetData())
+		w.Write(data.GetData()) // nolint: errcheck
```
Comment (phiggins) [inline components/automate-gateway/gateway/services.go]: It's not really your responsibility to fix this in this PR, but this seems like kind of a big deal...

Comment (tylercloke) [reply components/automate-gateway/gateway/services.go]: Now we know they are there :) Let's make a follow up ticket if we wanna error handle all of that!

**Pull request merged**